terminus/models/block.py

```diff
@@ -30,3 +31,8 @@ def square(origin, size):
     def accept(self, generator):
         generator.start_block(self)
         generator.end_block(self)
+
+    def bounding_box(self):
+        box_origin = self.origin + Point.min(self.vertices)
```

Comment:
Here you can apply the same method you used before: collect a set of bounding boxes (derived from the vertices) and do a `BoundingBox.from_boxes` to merge them all.